README.rst

```diff
@@ -1,43 +1,111 @@
+* Extensive test coverage.
+
+
+Usage
+-----
+The most straightforward way is to use the :func:`dateparser.parse` function,
+that wraps around most of the functionality in the module.
+
+.. automodule:: dateparser.__init__
+   :members: parse
+
+
+Popular Formats
++++++++++++++++
+
+    >>> import dateparser
+    >>> dateparser.parse('12/12/12')
+    datetime.datetime(2012, 12, 12, 0, 0)
+    >>> dateparser.parse(u'Fri, 12 Dec 2014 10:55:50')
+    datetime.datetime(2014, 12, 12, 10, 55, 50)
+    >>> dateparser.parse(u'Martes 21 de Octubre de 2014')  # Spanish
+    datetime.datetime(2014, 10, 21, 0, 0)
+    >>> dateparser.parse(u'Le 11 Décembre 2014 à 09:00')  # French
+    datetime.datetime(2014, 12, 11, 9, 0)
+    >>> dateparser.parse(u'13 января 2015 г. в 13:34')  # Russian
+    datetime.datetime(2015, 1, 13, 13, 34)
+    >>> dateparser.parse(u'1 เดือนตุลาคม 2005, 1:00 AM')  # Thai
+    datetime.datetime(2005, 10, 1, 1, 0)
+
+This will try to parse a date from the given string, attempting to
+detect the language each time.
+
+If you know beforehand the languages that the date should be, you can skip
+language detection specifying them using the ``languages`` argument::
 
-If you have needed to parse dates before, you probably have used the date
-parser of the dateutil_ module.
-We built this library on top of it, adding a few features:
+    >>> dateparser.parse('2015, Ago 15, 1:08 pm', languages=['pt', 'es'])
+    datetime.datetime(2015, 8, 15, 13, 8)
 
-* dateparser support dates in languages other than English
-* in fact, it can detect the language automatically
-* it can give you the date for text like: ``'1 min ago'``, ``'2 weeks ago'``, ``'3 months, 1 weeks and 1 day ago'``, etc
+If you know the possible formats that the date will be, you can
+use the ``date_formats`` argument::
 
+    >>> dateparser.parse(u'22 Décembre 2010', date_formats=['%d %B %Y'])
+    datetime.datetime(2010, 12, 22, 0, 0)
 
-The goal is to support the common date formats used in websites all around the world.
+
+Contextual Dates
+++++++++++++++++
+
+    >>> parse('1 hour ago')
+    datetime.datetime(2012, 12, 21, 23, 0)
+    >>> parse(u'Il ya 2 heures')  # French
+    datetime.datetime(2012, 12, 21, 22, 0)
+    >>> parse(u'1 anno 2 mesi')  # Italian
+    datetime.datetime(2011, 10, 22, 0, 0)
+    >>> parse(u'yaklaşık 23 saat önce')  # Russian
+    datetime.datetime(2012, 12, 21, 1, 0)
+    >>> parse(u'Hace una semana')  # Spanish
+    datetime.datetime(2012, 12, 15, 0, 0)
+    >>> parse(u'2小时前')  # Chinese
+    datetime.datetime(2012, 12, 21, 22, 0)
+
+.. note:: Testing above code might return different values for you depending on your environment's current date time. To reproduce exactly above results, you can patch your environment as follows::
+
+    >>> from dateparser.date import freshness_date_parser
+    >>> patch.object(freshness_date_parser, 'now', datetime(2012, 12, 22, 0, 0)).start()
```

Comment:
As strange as it sounds, not everybody can be familiar with mocking mechanism. Let's just use more recent date (like 1st June 2015) and state that user would get different results depending on the current date and time. Also let's provide translations for the given dates (f.ex "French (2 hours ago)"). And "yaklaşık 23 saat önce" is not russian :smile: